pkg/distro/rhel/images.go

```diff
@@ -223,8 +224,17 @@ func osCustomizations(
 		osc.OpenSCAPRemediationConfig = remediationConfig
 	}
 
-	osc.ShellInit = imageConfig.ShellInit
+	if options.Subscription != nil {
+		if rhsmConfig, exists := imageConfig.RHSMConfig[subscription.RHSMConfigWithSubscription]; exists {
+			osc.RHSMConfig = rhsmConfig
+		}
+	} else {
+		if rhsmConfig, exists := imageConfig.RHSMConfig[subscription.RHSMConfigNoSubscription]; exists {
+			osc.RHSMConfig = rhsmConfig
+		}
+	}
```

Comment:
Good change.  This makes more sense to me too.

Reply:
I wonder if this could be written slightly less nested, maybe something like:
```go
	var subscriptionStatus subscription.RHSMStatus
	if options.Subscription != nil {
		subscriptionStatus = subscription.RHSMConfigWithSubscription
	} else {
		subscriptionStatus = subscription.RHSMConfigNoSubscription
	}
	if rhsmConfig, exists := imageConfig.RHSMConfig[subscriptionStatus]; exists {
		osc.RHSMConfig = rhsmConfig
	}
```

Reply:
I reduced the nesting in a separate commit.